src/mem3/src/mem3_shards.erl

```diff
@@ -406,6 +458,43 @@ cache_clear(St) ->
     true = ets:delete_all_objects(?ATIMES),
     St#st{cur_size=0}.
 
+maybe_spawn_shard_writer(DbName, Shards) ->
+    case ets:member(?OPENERS, DbName) of
+        true ->
+            ignore;
+        false ->
+            spawn_shard_writer(DbName, Shards)
+    end.
+
+spawn_shard_writer(DbName, Shards) ->
+    erlang:spawn(fun() -> shard_writer(DbName, Shards) end).
+
+shard_writer(DbName, Shards) ->
+    try
+        receive
+            write ->
+                true = ets:insert(?SHARDS, Shards);
+            cancel ->
+                ok
+        after ?WRITE_IDLE_TIMEOUT ->
+            ok
+        end
+    after
+        true = ets:delete(?OPENERS, {DbName, self()})
+    end.
+
+flush_write(DbName, Writer) ->
+    Ref = erlang:monitor(process, Writer),
+    Writer ! write,
+    receive
+        {'DOWN', Ref, _, _, normal} ->
+            ok;
+        {'DOWN', Ref, _, _, Error} ->
+            erlang:exit({mem3_shards_bad_write, Error})
+    after ?WRITE_TIMEOUT ->
+        erlang:exit({mem3_shards_write_timeout, DbName})
```

Comment:
Since process is not linked and we time out. Should kill it here a well perhaps, to make sure it is not left lingering around?

Another crazy idea I had. what if we to link to writer in `flush_write` or maybe as soon as gen_server gets them `handle_cast({cache_insert*...)`. Idea being gen_server doesn't want to lose the writer process and now it kind of "owns" it and wants to know if it crashes un-expectedly. If we do that then we'd skip the `{'DOWN', Ref, _, _, Error}` clause. But we'd have an extra link to mange, so... `¯\_(ツ)_/¯`